File: erubit.platform/src/main/java/a/erubit/platform/android/CourseTrainingFragment.java
package a.erubit.platform.android;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import a.erubit.platform.interaction.InteractionManager;
import a.erubit.platform.R;
import a.erubit.platform.course.Course;
import a.erubit.platform.course.CourseManager;
import a.erubit.platform.course.Lesson;

public class CourseTrainingFragment extends TrainingFragment {
    private Course mCourse;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        if (null == getArguments())
            mCourse = null;
        else
            mCourse = CourseManager.i().getCourse(getArguments().getString("id"));

        return super.onCreateView(inflater, container, savedInstanceState);
    }

    @Override
    public View getNextInteractionView() {
        Lesson lesson;
        if (mCourse != null)
            lesson = CourseManager.i().getNextLesson(getContext(), mCourse);
        else
            lesson = CourseManager.i().getNextLesson(getContext());

        if (lesson == null)
            return null;

        View v = InteractionManager.i().getInteractionView(getContext(), lesson, this);
        v.findViewById(R.id.quickButtonBar).setVisibility(View.GONE);

        return v;
    }
}

File: erubit.platform/src/main/java/a/erubit/platform/CourseTrainingFragment.kt
package a.erubit.platform

import a.erubit.platform.course.Course
import a.erubit.platform.course.CourseManager
import a.erubit.platform.interaction.InteractionManager

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup


class CourseTrainingFragment : TrainingFragment() {
	private var mCourse: Course? = null

	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
		mCourse = if (null == arguments) null else CourseManager.i().getCourse(arguments!!.getString("id"))
		return super.onCreateView(inflater, container, savedInstanceState)
	}

	override val nextInteractionView: View?
		get() {
			val context = this.context ?: return null

			val lesson = (if (mCourse != null) CourseManager.i().getNextLesson(mCourse!!) else CourseManager.i().getNextLesson())
				?: return null

			return InteractionManager.i().getInteractionView(context, lesson, this) ?: return null
		}
}
